packages/flutter_test/lib/src/controller.dart

```diff
@@ -95,6 +104,104 @@ class SemanticsController {
+  /// related node does not support the given `action`, then the closest
+  /// descendent of that node that does support the `action` is returned.
+  ///
+  /// Will throw a [StateError] if:
+  /// * No results are returned from `finder`
+  /// * More than result is returned from `finder`
+  /// * The [SemanticsNode] of `finder` or its descendant nodes do not support
+  ///   `action`
+  ///
+  /// See also:
+  /// * [find]
+  /// * [findWithAnyAction]
+  SemanticsNode findWithAction(Finder finder, SemanticsAction action) {
+    final SemanticsNode found = find(finder);
+    SemanticsNode? actionable;
+    _traverse(found, (SemanticsNode node) {
+      if (node.getSemanticsData().hasAction(action)) {
+        actionable = node;
+        return true;
+      }
+      return false;
+    });
+
+    if (actionable == null) {
+      throw StateError(
+        'The given finder did not find a node that supports (or has '
+        'a descendant that supports) $action.\n'
+        'Found Node: $found'
+      );
+    }
+    return actionable!;
+  }
+
+  /// Attemps to find a [SemanticsNode] with any of the given `actions` as
+  /// close to the [SemanticsNode] of `finder` as possible.
+  ///
+  /// The [SemanticsNode] directly related to an [Element] is not necessarily
+  /// the one that can be acted upon. This method finds the directly related
+  /// node and checks whether it supports any of the given `actions`. If the
+  /// directly related node does not support any of the given `actions`, then
+  /// the closest descendent of that node that does support at least one of
+  /// the `actions` is returned.
+  ///
+  /// Will throw a [StateError] if:
+  /// * No results are returned from `finder`
+  /// * More than result is returned from `finder`
+  /// * The [SemanticsNode] of `finder` or its descendant nodes do not support
+  ///   at least one action in `actions`
+  ///
+  /// See also:
+  /// * [find]
+  /// * [findWithAction]
+  SemanticsNode findWithAnyAction(Finder finder, List<SemanticsAction> actions) {
+    final int actionInt = actions.fold<int>(0,
+      (int prev, SemanticsAction act) => prev | act.index);
+
+    final SemanticsNode found = find(finder);
+    SemanticsNode? actionable;
+    _traverse(found, (SemanticsNode node) {
+      if (node.getSemanticsData().actions & actionInt != 0) {
+        actionable = node;
+        return true;
+      }
+      return false;
+    });
+
+    if (actionable == null) {
+      throw StateError(
+        'The given finder did not find a node that supports any (or has '
+        'a descendant that supports any) of the following actions:\n'
+        '[\n  ${actions.join(',\n  ')}\n]\n'
+        'Found Node: $found'
+      );
+    }
+
+    return actionable!;
+  }
+
+  /// Finds a node related to `finder` that can be scrolled semantically with:
+  /// [scrollUp], [scrollDown], [scrollLeft], or [scrollRight].
+  SemanticsNode findScrollable(Finder finder) {
```

Comment:
Haven't thought about this too much, but could these `find` methods be expressed as `Finder`s so they are combinable with the other Finders? E.g. how would I find a scrollable widget that is a descendant of a widget of type container?

Reply:
Alternatively wondering if these find* methods should be private and not part of the public API